Jenkinsfile

```
@@ -205,110 +205,80 @@ stage('Configure') {
-							condition: TestCondition.ON_DEMAND),
-					new EsLocalBuildEnvironment(versionRange: '[6.8,7.0)', mavenProfile: 'elasticsearch-6.8',
-							condition: TestCondition.AFTER_MERGE),
+					new EsLocalBuildEnvironment(version: '6.7.2', condition: TestCondition.ON_DEMAND),
+					new EsLocalBuildEnvironment(version: '6.8.22', condition: TestCondition.AFTER_MERGE),
 					// Not testing 7.0/7.1/7.2 to make the build quicker.
 					// The only difference with 7.3+ is they have a bug in their BigInteger support.
-					new EsLocalBuildEnvironment(versionRange: '[7.0,7.3)', mavenProfile: 'elasticsearch-7.0',
-							condition: TestCondition.ON_DEMAND),
-					new EsLocalBuildEnvironment(versionRange: '[7.3,7.7)', mavenProfile: 'elasticsearch-7.3',
-							condition: TestCondition.AFTER_MERGE),
+					new EsLocalBuildEnvironment(version: '7.2.1', condition: TestCondition.ON_DEMAND),
+					new EsLocalBuildEnvironment(version: '7.6.2', condition: TestCondition.AFTER_MERGE),
 					// Not testing 7.7 to make the build quicker.
 					// The only difference with 7.7+ is how we create templates for tests.
-					new EsLocalBuildEnvironment(versionRange: '[7.7,7.8)', mavenProfile: 'elasticsearch-7.7',
-							condition: TestCondition.ON_DEMAND),
+					new EsLocalBuildEnvironment(version: '7.7.1', condition: TestCondition.ON_DEMAND),
 					// Not testing 7.9 to make the build quicker.
 					// The only difference with 7.10+ is an additional test for exists on null values,
 					// which is disabled on 7.10 but enabled on all older versions (not just 7.9).
-					new EsLocalBuildEnvironment(versionRange: '[7.8,7.10)', mavenProfile: 'elasticsearch-7.8',
-							condition: TestCondition.AFTER_MERGE),
-					new EsLocalBuildEnvironment(versionRange: '[7.10,7.11)', mavenProfile: 'elasticsearch-7.10',
-							condition: TestCondition.AFTER_MERGE),
+					new EsLocalBuildEnvironment(version: '7.9.3', condition: TestCondition.AFTER_MERGE),
+					new EsLocalBuildEnvironment(version: '7.10.1', condition: TestCondition.AFTER_MERGE),
 					// Not testing 7.11 to make the build quicker.
 					// The only difference with 7.12+ is that wildcard predicates on analyzed fields get their pattern normalized,
 					// and that was deemed a bug: https://github.com/elastic/elasticsearch/pull/53127
-					new EsLocalBuildEnvironment(versionRange: '[7.11,7.12)', mavenProfile: 'elasticsearch-7.11',
-							condition: TestCondition.ON_DEMAND),
-					new EsLocalBuildEnvironment(versionRange: '[7.12,8.0)', mavenProfile: 'elasticsearch-7.12',
-							condition: TestCondition.AFTER_MERGE),
+					new EsLocalBuildEnvironment(version: '7.11.2', condition: TestCondition.ON_DEMAND),
+					new EsLocalBuildEnvironment(version: '7.12.1', condition: TestCondition.ON_DEMAND),
+					new EsLocalBuildEnvironment(version: '7.13.2', condition: TestCondition.ON_DEMAND),
+					new EsLocalBuildEnvironment(version: '7.16.3', condition: TestCondition.ON_DEMAND),
```

Let's also move comments from the POM:

```suggestion
					new EsLocalBuildEnvironment(version: '7.13.2', condition: TestCondition.ON_DEMAND),
					// 7.14 and 7.15 have annoying bugs that make almost all of our test suite fail,
					// so we don't test them
					// See https://hibernate.atlassian.net/browse/HSEARCH-4340
					new EsLocalBuildEnvironment(version: '7.16.3', condition: TestCondition.ON_DEMAND),
```